src/panels/config/cloud/login/cloud-login.ts

```diff
@@ -1,217 +1,115 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, query } from "lit/decorators";
 import { fireEvent } from "../../../../common/dom/fire_event";
-import { navigate } from "../../../../common/navigate";
 import "../../../../components/buttons/ha-progress-button";
 import "../../../../components/ha-alert";
 import "../../../../components/ha-card";
-import "../../../../components/ha-icon-next";
-import "../../../../components/ha-list-item";
+import "../../../../components/ha-button";
 import "../../../../components/ha-password-field";
-import "../../../../components/ha-button-menu";
 import type { HaPasswordField } from "../../../../components/ha-password-field";
 import "../../../../components/ha-textfield";
 import type { HaTextField } from "../../../../components/ha-textfield";
-import { setAssistPipelinePreferred } from "../../../../data/assist_pipeline";
-import { cloudLogin, removeCloudData } from "../../../../data/cloud";
-import {
-  showAlertDialog,
-  showConfirmationDialog,
-  showPromptDialog,
-} from "../../../../dialogs/generic/show-dialog-box";
-import "../../../../layouts/hass-subpage";
 import { haStyle } from "../../../../resources/styles";
-import type { HomeAssistant } from "../../../../types";
-import "../../ha-config-section";
-import { showSupportPackageDialog } from "../account/show-dialog-cloud-support-package";
-import { showCloudAlreadyConnectedDialog } from "../dialog-cloud-already-connected/show-dialog-cloud-already-connected";
+import type { LocalizeFunc } from "../../../../common/translations/localize";
 
 @customElement("cloud-login")
 export class CloudLogin extends LitElement {
-  @property({ attribute: false }) public hass!: HomeAssistant;
-
-  @property({ attribute: "is-wide", type: Boolean }) public isWide = false;
-
-  @property({ type: Boolean }) public narrow = false;
-
   @property() public email?: string;
 
-  @property({ attribute: false }) public flashMessage?: string;
+  @property() public password?: string;
 
-  @state() private _password?: string;
+  @property({ attribute: false }) public localize!: LocalizeFunc;
 
-  @state() private _requestInProgress = false;
+  @property({ type: Boolean, attribute: "in-progress" }) public inProgress =
+    false;
 
-  @state() private _error?: string;
+  @property() public error?: string;
 
-  @state() private _checkConnection = true;
+  @property({ attribute: "translation-key-panel" }) public translationKeyPanel:
+    | "page-onboarding.restore.ha-cloud"
+    | "config.cloud" = "config.cloud";
 
-  @query("#email", true) private _emailField!: HaTextField;
+  @property({ type: Boolean, attribute: "card-less" }) public cardLess = false;
+
+  @query("#email", true) public _emailField!: HaTextField;
 
   @query("#password", true) private _passwordField!: HaPasswordField;
 
   protected render(): TemplateResult {
+    if (this.cardLess) {
+      return this._renderLoginForm();
+    }
```

Comment:
Why not just wrap the element in a `ha-card` if the parent wants that? This element should not be responsible for rendering the card.